src/jaxsim/api/data.py

```diff
@@ -185,16 +185,19 @@ def build(
             )
         )
 
-        if isinstance(model.contact_model, SoftContacts):
-            contacts_params = (
-                contacts_params
-                if contacts_params is not None
+        contacts_params = (
+            contacts_params
+            if contacts_params is not None
+            else (
+                model.contact_model.parameters
+                if not isinstance(
+                    model.contact_model, jaxsim.rbda.contacts.SoftContacts
+                )
                 else js.contact.estimate_good_soft_contacts_parameters(
                     model=model, standard_gravity=standard_gravity
                 )
             )
```

Comment:
Maybe we could make the logic a bit more linear instead of nesting two ternary operators. Note that the suggestion has to be reformatted as I cannot comment on deleted lines
```suggestion
if contacts_params is None:
    if isinstance(model.contact_model, jaxsim.rbda.contacts.SoftContacts):
        contacts_params = js.contact.estimate_good_soft_contacts_parameters(
            model=model, standard_gravity=standard_gravity
        )
    else:
        contacts_params = model.contact_model.parameters
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/247/commits/4fc99ea7ac62fe3b54f08471261b00dd730f2250